libs/core/type_support/include/hpx/type_support/is_contiguous_iterator.hpp

```diff
@@ -0,0 +1,84 @@
+//  Copyright (c) 2007-2022 Hartmut Kaiser
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <array>
+#include <string>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+namespace hpx::traits {
+    ///////////////////////////////////////////////////////////////////////////
+    // Iterators are contiguous if they are pointers (without concepts we have
+    // no generic way of determining whether an iterator is contiguous)
+
+    namespace detail {
+
+        // Iterators returned from std::vector are contiguous (by definition)
+        template <typename Iter>
+        using iter_value_type_t =
```

Comment:
FWIW, the standard calls this `std::iter_value_t`.

Reply:
Good to know, I want to avoid double defining it since it is already in `iterator_support/is_iterator.hpp`